root/static/scripts/edit/URLCleanup.js

```diff
@@ -321,12 +322,13 @@ const CLEANUPS = {
     type: LINK_TYPES.discogs,
     clean: function (url) {
       url = url.replace(/\/viewimages\?release=([0-9]*)/, "/release/$1");
-      url = url.replace(/^https?:\/\/(?:[^.]+\.)?discogs\.com\/(?:.*\/)?(user\/[^\/#?]+|(?:artist|release|master(?:\/view)?|label)\/[0-9]+)(?:[\/#?-].*)?$/, "https://www.discogs.com/$1");
+      url = url.replace(/\/composition\/([^-]+-[^-]+-[^-]+-[^-]+-[^-]+).*/, "/composition/$1");
+      url = url.replace(/^https?:\/\/(?:[^.]+\.)?discogs\.com\/(?:.*\/)?(user|composition\/[^\/#?]+|(?:artist|release|master(?:\/view)?|label)\/[0-9]+)(?:[\/#?-].*)?$/, "https://www.discogs.com/$1");
```

Comment:
Main issue is you broke the pattern `user\/[^\/#?]+`. Plus, you could merge these two added lines by moving the _capture group_ for UUID from the first line into the RegExp on the second line.

Indeed, the purpose of this line is twofold: first, normalizing the heading scheme to `https` and the subdomain to `www`; then, trimming extraneous trailing characters. The preceding line you added is somehow duplicating the second purpose of this line.